app/services/xccdf_report_parser.rb

```diff
@@ -58,10 +58,11 @@ def save_all
 
   def save_rule_results
     RuleResult.import!(
-      rule_results_rule_ids.zip(@oscap_parser.rule_results.map(&:result))
-      .each_with_object([]) do |rule_result, rule_results|
+      @oscap_parser.rule_results.each_with_object([]) do |rule_result, rule_results|
         rule_results << RuleResult.new(
-          host: @host, rule_id: rule_result[0], result: rule_result[1],
+          host: @host,
+          rule_id: rule_results_rule_ids[rule_result.id],
```

Comment:
This is where the hash is used and where ref_id is matched between the DB and xml results.

Reply:
Shouldn't this field be `rule_results_rule_ids[rule_result.id].id`? Seems to work like this just passing the object though, according to the test.

Reply:
the hash has the structure `{<ref_id> => <id>}`, so when you call `rule_results_rule_ids[rule_result.id]` it returns the id value.